### DISCUSSION TITLE
BirdNET-Pi with static IP inside a LAN

### DISCUSSION THREAD

**Discussion by DD4WH** [mcguirepr89/BirdNET-Pi#167] [Q&A]
I have a BirdNET-Pi inside a large LAN network which requires a static IP adress.
I altered /etc/dhcpcd.conf and inserted the following:
`auto eth0`
`interface eth0`
`static ip_adress=MY.IP.ADD.RESS:24`
`static routers=MY.STANDARD.GATE.WAY`
`static domain_name_servers=MY.DNS.SER.VERS`
Because connection to the Pi was only possible by putty by that time, I altered `birdnet.conf`
and changed this:
`BIRDNETPI_URL=https://MY.IP.ADD.RESS`
`BIRDNETPI_URL=https://MY.IP.ADD.RESS:8080`
`BIRDNETPI_URL=https://MY.IP.ADD.RESS:8888`
Then I did the same in `sudo nano /etc/caddy/Caddyfile`
and changed/added the same IPs

After that, I can now access the Log and the Web Terminal page, but not the main page.

Do I need to change something else?

**Comment by mcguirepr89** [comment]
@DD4WH 
**HEAVILY Edited a few times!**_I was wrong about everything I said a few times after looking back through the code and doing more testing._

For a static IP address using HTTPS, do everything you have but choose different ports (above 6000) since 8080 and 8888 are used in for GoTTY.

I chose ports 9090 and 9898 because they're easy to type, but I've verified that works. Sorry for all the crappy first responses!!!

**IMPORTANT NOTE**
If you are using HTTPS for security purposes, you'll want to disable GoTTY's responses on port 8888 so that Caddy handles the Basic Auth and TLS. Do this by editing the `templates/web_terminal.service` file's `ExecStart` line from:
```
ExecStart=/usr/local/bin/gotty -w -p 8888 --title-format "BirdNET-Pi Terminal" bash
```
to
```
ExecStart=/usr/local/bin/gotty --address localhost -w -p 8888 --title-format "BirdNET-Pi Terminal" bash
```
and then reloading the service file
```
sudo systemctl daemon-reload
```
then restart the Web Terminal service in "Tools" > "Services" or disabled it altogether.

Yikes I need to be more thoughtful before responding.

**Reply by mcguirepr89** [reply]
Be sure to run ```sudo systemctl reload caddy``` if you're editing the Caddyfile manually.

**Comment by DD4WH** [comment]
Thank you very much, Patrick for your detailed and carefully tested reply and suggestions!
At the moment, I am still struggling with getting the Pi to have internet access inside the network (using a proxy server), so I will adress the various problems I have setp-by-step and come back to you later! Thanks again!

**Reply by DD4WH** [reply]
I think I am now ready for adressing this issue again. This is where I am standing now:
* the BirdNET-Pi has a static IP adress (eth0) inside our LAN network and is recognized by the network (I can reach it via SSH)
* we configured the Pi to use our proxy server for its access to the internet --> with `curl https://curl.se` I can see a listing of the HTML code of the page, so it seems the Pi has access to the internet
* the BirdNET-Pi webpage is accessible inside our intranet by simply using `https://BIRDNET.INTERNAL.IP.ADRESS`

So far, so good!

Now I have two small issues to solve;
* when accessing the BirdNET-Pi by `https://BIRDNET.INTERNAL.IP.ADRESS`, the browser complains about a non-private connection, obviously because there is no certificate!? --> `NET::ERR_CERT_AUTHORITY_INVALID`   However, I think, the connection is still encrypted, i.e. a https connection?
* I am a little confused about the port issue mentioned above. Your recommendation is to put this into the Settings ?
`BirdNET-PI URL https://BIRDNET.INTERNAL.IP.ADRESS`
`BirdNET-PI Log URL http://BIRDNET.INTERNAL.IP.ADRESS:9090`
`BirdNET-PI Web Terminal http://BIRDNET.INTERNAL.IP.ADRESS:9898`
Or do I need to modify the Caddyfile ? At this point now, I have access to the Settings, so I can easily alter the settings the "official" way.

**Reply by mcguirepr89** [reply]
>when accessing the BirdNET-Pi by https://BIRDNET.INTERNAL.IP.ADRESS, the browser complains about a non-private connection, obviously because there is no certificate!? --> NET::ERR_CERT_AUTHORITY_INVALID However, I think, the connection is still encrypted, i.e. a https connection?

You are correct that the connection is still encrypted, but there is indeed a certificate, it is not, however 'valid'. The complaint from the web browser is letting you know that the website that you are visiting issued its own SSL Certificate, or in other words, it is letting you know that the SSL certificate that the site you're visiting did not have a third-party Certificate Authority (CA) sign the certificate, but instead signed it itself. Your browser warns you about this so that you understand that no Certificate Authority has verified the ownership of the website. Since you're the owner and can verify that the SSL certificate your web browser received was indeed issued and signed by Caddy, that is tantamount to having it signed by a third-party CA for ownership and verification purposes. As far as encryption, it is the same level of encryption that you would obtain if you were to register a FQDN and purchase an SSL certificate from a CA.

>I am a little confused about the port issue mentioned above. Your recommendation is to put this into the Settings ?
BirdNET-PI URL https://BIRDNET.INTERNAL.IP.ADRESS
BirdNET-PI Log URL http://BIRDNET.INTERNAL.IP.ADRESS:9090
BirdNET-PI Web Terminal http://BIRDNET.INTERNAL.IP.ADRESS:9898
Or do I need to modify the Caddyfile ? At this point now, I have access to the Settings, so I can easily alter the settings the "official" way.

For your purposes, use the "Advanced Settings" page in the web interface and set the URLs as described above. This will reverse_proxy requests to the log and web terminal through Caddy so that Caddy can add SSL certificates to that connection.

I'm reworking how the Custom URLs are handled and very soon there will only be one custom URL for the entire installation, so this is kind of a temporary frustration at this point. :)

My best!

**Reply by DD4WH** [reply]
Thanks Patrick, for your quick response and the clear explanations! 
So, my mistake is that I have to add `https`, NOT `http` for the Log and Web Terminal URLs in the Advanced Settings ?

**Reply by mcguirepr89** [reply]
I'm sorry I didn't even notice the differences in URL protocols, but _yes_ use `https://` for all three URLs

**Reply by DD4WH** [reply]
When I change the Settings and type `https://birdnet.internal.ip.adress:9898/`, the following error occurs:
`SSL_ERROR_RX_RECORD_TOO_LONG`
I have not (yet) stopped the Gotty service as you recommended above, maybe thats the cause?

**Reply by mcguirepr89** [reply]
Hmm I've never seen nor heard of that error before.

Restart the BirdNET Log / Web Terminal service and see if that solves things. :thinking:

**Reply by DD4WH** [reply]
Thanks Patrick, for your help!
I tried a few things, but the error remains.
However, that is not too relevant, I think. If I can reach the Pi via SSH, I would not need the Web Terminal.
Can I still reach the log page via port 8080, if I disable GoTTY for port 8888, as you described above ? [Sorry for these silly questions, this network stuff is not so easy for me to grasp]

**Reply by mcguirepr89** [reply]
I'm sorry I can't be of more help with this. Are you trying to reach the Log and Web terminal from the main site using the buttons (i.e., clicking "View Log" and "Tools" > "Web Terminal"), or are you typing the addresses (and ports) into the web browser address bar to reach things?

**Comment by AdamM68** [comment]
following the subject ....
my birdnetpi is runnig fine so far for couple of months but it is time to switch my rPi from DHCP adress to static one.
I used ssh connection to my rPi and edited dhcpcd.conf as follow:
`birdnetpi:~/BirdNET-Pi$ sudo nano /etc/dhcpcd.conf

# A sample configuration for dhcpcd.
# … all comments removed for clarity
hostname
clientid
persistent
require dhcp_server_identifier
slaac private
static routers=192.168.0.1

interface eth0
static ip_address=192.168.1.114/24
static routers=192.168.1.1
static domain_name_servers=192.168.1.1

interface wlan0
static ip_address=192.168.1.113/24
static routers=192.168.1.1
static domain_name_servers=192.168.1.1`

Now... my wifi connection (wlan0) is working fine and rPi is using static ip .113 but eth0 is still using dhcp adress (.229) - it is not using static .114

Whats wrong ?
should I change something more or the cinfiguration file dhcpcd.conf is wrong ?

**Reply by AdamM68** [reply]
found an information that only 1 interface can be declared in dhcpcd.conf ..??? is it true ?

[https://forums.raspberrypi.com/viewtopic.php?t=140252](url)
